### PR TITLE
Session garbage collection configuration

### PR DESCRIPTION
**Pull request by victorpetryk**
## Purpose of this pull request

This pull request (PR) adds a new section about configuring garbage collection in the env.php file.

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.4/config-guide/sessions.html

## Links to Magento source code

- https://github.com/magento/magento2/issues/26308

whastnew
Added a new section about [configuring garbage collection](https://devdocs.magento.com/guides/v2.4/config-guide/sessions.html) in Magento to clean up expired sessions.

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
@victorpetryk thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by jeff-matthews** [comment]
@victorpetryk, does this apply to both the 2.4.x and 2.3.x release lines or 2.4.x only?

File: src/_includes/config/locate-session.md [hunk 1]
```diff
@@ -27,3 +27,24 @@ The preceding example stores session files in `/var/www/session`
 ### `php.ini` example {#session-where-phpini}
 
 As a user with `root` privileges, open your `php.ini` file and search for the value of `session.save_path`. This identifies where sessions are stored.
+
+## Garbage collection configuration {#session-gc}
+
+To clean up expired sessions Magento application calls `gc` (_garbage collection_) handler randomly according to probability which is calculated by `gc_probability / gc_divisor`. For example, if we set these directives to `1/100` respectively, it means a probability of `1%` (_probability of one call of garbage collection per 100 requests_).
```
Comment (jeff-matthews) [inline src/_includes/config/locate-session.md:33]:
```suggestion
To clean up expired sessions, Magento calls the `gc` (_garbage collection_) handler randomly according to a probability that is calculated by the `gc_probability / gc_divisor` directive. For example, if you set these directives to `1/100` respectively, it means a probability of `1%` (_probability of one call of garbage collection per 100 requests_).
```

File: src/_includes/config/locate-session.md [hunk 2]
```diff
@@ -27,3 +27,24 @@ The preceding example stores session files in `/var/www/session`
...
+
+The garbage collection handler uses the `gc_maxlifetime` directive - the number of seconds after which the sessions will be seen as _garbage_ and potentially cleaned up.
```
Comment (jeff-matthews) [inline src/_includes/config/locate-session.md:35]:
```suggestion
The garbage collection handler uses the `gc_maxlifetime` directive—the number of seconds after which the sessions will be seen as _garbage_ and potentially cleaned up.
```

File: src/_includes/config/locate-session.md [hunk 3]
```diff
@@ -27,3 +27,24 @@ The preceding example stores session files in `/var/www/session`
...
+
+On some operating systems (Debian/Ubuntu) `session.gc_probability` set to `0` by default, and the garbage collection handler will never be executed.
```
Comment (jeff-matthews) [inline src/_includes/config/locate-session.md:37]:
```suggestion
On some operating systems (Debian/Ubuntu), the default `session.gc_probability` directive is `0`, which prevents the garbage collection handler from running.
```

File: src/_includes/config/locate-session.md [hunk 4]
```diff
@@ -27,3 +27,24 @@ The preceding example stores session files in `/var/www/session`
...
+
+`session.gc_` directives from the `php.ini` file can be overwritten in `<magento_root>/app/etc/env.php`:
```
Comment (jeff-matthews) [inline src/_includes/config/locate-session.md:39]:
```suggestion
You can overwrite the `session.gc_` directives from the `php.ini` file in the `<magento_root>/app/etc/env.php` file:
```

File: src/_includes/config/locate-session.md [hunk 5]
```diff
@@ -27,3 +27,24 @@ The preceding example stores session files in `/var/www/session`
...
+```php
+ 'session' =>
+    array (
+      'save' => 'db',
+      'gc_probability' => 1,
+      'gc_divisor' => 1000,
+      'gc_maxlifetime' => 1440
+ ),
+```
+The configuration depends on the traffic and specific needs of a certain Magento application.
```
Comment (jeff-matthews) [inline src/_includes/config/locate-session.md:50]:
```suggestion

The configuration depends on the traffic and specific needs of a certain Magento application.
```

**Review by jeff-matthews — CHANGES_REQUESTED** [review]

**Comment by victorpetryk** [comment]
> @victorpetryk, does this apply to both the 2.4.x and 2.3.x release lines or 2.4.x only?

@jeff-matthews This applies to 2.4.x only as changes to configure garbage collection from env.php file will be delivered to 2.4.2

**Comment by jeff-matthews** [comment]
Thanks for confirming @victorpetryk. 

Since it applies to 2.4.x only, I need you to do the following:

1. Replace the symlinked `src/guides/v2.4/config-guide/sessions.md` file with a copy of the `src/guides/v2.3/config-guide/sessions.md` file.
1. Replace the `include` reference (`{% include config/locate-session.md %}`) from the new 2.4 file with the revised text of your new section (https://github.com/magento/devdocs/pull/8053/commits/e436d7744e420daa500aa4233f1022786d457140). We don't use this include file anywhere else inn devdocs, so I don't think we even need it.
1. Revert all changes to the `src/_includes/config/locate-session.md` file. This will be sued for 2.3 only and should not contain information about garbage collection.

I understand that may be tedious, so if you prefer, I would be happy to create a copy of your branch and do it myself. Just let me know what you prefer.

**Comment by victorpetryk** [comment]
@jeff-matthews Let me ask PO about the necessity to backport a fix about GC to the 2.3.x release line. And if so the devdocs will need to be updated for 2.3.x as well.

**Comment by victorpetryk** [comment]
@jeff-matthews I've created a backported fix for the 2.3.x release line. So we can continue to proceed with this PR.

**Comment by jeff-matthews** [comment]
Sounds good @victorpetryk. Thanks!

**Comment by jeff-matthews** [comment]
running tests

**Review by jeff-matthews — APPROVED** [review]

**Pull request merged**